docs/DesignDoc.md

```diff
@@ -32,16 +32,69 @@ This section describes the features of the application.
 
 ## Architecture and Design
 
-This section describes the application architecture.
-
 ### Software Architecture
-> _Place a architectural diagram here._
-> _Describe your software architecture._
+![architectural-pattern](..%2Fassets%2Farchitectural-pattern.png)
+
+For the software architecture design diagram of “TigerBets”, we adopt a layered approach. From top to bottom, it contains four layers:
+
+**Presentation Layer**:
+
+- Contains the User Interface (UI) and Simple Visual Charts.
+
+**Application Layer**:
+
+- Consists of Business Logic, Application Controllers for Authentication (Auth), Events, Accounts, Betting, and Analytics.
+
+**Domain Layer**:
+
+- Involves Business Rules and Data Analytics, along with Entities such as Event, User, and Bet.
+
+**Persistence Layer:**
+
+- Includes the Database for Authentication (Auth), User information, Events, Bets, and the Leaderboard.
+
+Each layer communicates information down to the next, indicating that the upper layers rely on the data and services provided by the lower ones. This architectural design emphasizes separation of concerns, with each layer focusing on specific functionalities to enhance the maintainability and scalability of the software.
 
 
 ### Use Cases
-> _Place a use case diagram here._
-> _Describe your use case diagram._
+ 
+![use-case-diagram](..%2Fassets%2Fuse-case-diagram.png)
+ 
+***
+
+**Use Case Detailed Descriptions**
+
+This use case diagram describes two actors in the TigerBets system: regular end users and system administrators. Each has their own functions and use cases. The diagram also illustrates the relationship between these two different actors. 
+
+------
```

Comment:
Considering we don't use separations in the other sections, we can remove it for consistency?